app.go

```diff
@@ -461,6 +467,25 @@ func (app *App) StopTimeout() time.Duration {
 	return app.stopTimeout
 }
 
+func (app *App) visualizeDOT() DotGraph {
+	// Memoize the graph
+	if app.dotgraph != nil {
+		return *app.dotgraph
+	}
+	b := &bytes.Buffer{}
+	if err := dig.Visualize(app.container, b); err != nil {
+		b.WriteString(`digraph {
```

Comment:
Instead of WriteString-ing a string into the buffer and then extracting it, we can just return it directly.

```
if err := ...; err != nil {
    return DotGraph(...)
}
```